src/cpp/src/cache_manager.hpp

```diff
@@ -223,14 +236,32 @@ class CacheManager {
                     key_dst_end_roi[0] = (key_dst_start_roi[0] = dst_block_id) + 1;
                     value_dst_end_roi[0] = (value_dst_start_roi[0] = dst_block_id) + 1;
 
-                    ov::Tensor key_src_cache_roi(m_key_cache[decoder_layer_id], key_src_start_roi, key_src_end_roi);
-                    ov::Tensor key_dst_cache_roi(m_key_cache[decoder_layer_id], key_dst_start_roi, key_dst_end_roi);
-
-                    ov::Tensor value_src_cache_roi(m_value_cache[decoder_layer_id], value_src_start_roi, value_src_end_roi);
-                    ov::Tensor value_dst_cache_roi(m_value_cache[decoder_layer_id], value_dst_start_roi, value_dst_end_roi);
-
-                    key_src_cache_roi.copy_to(key_dst_cache_roi);
-                    value_src_cache_roi.copy_to(value_dst_cache_roi);
+                    auto copy_one_block = [&](ov::Tensor& dst, const ov::Tensor& src, size_t src_start, size_t dst_start, size_t stride) {
+                        auto sub_byte_multipyer = sub_byte_data_type_multiplier(dst.get_element_type());
+                        const uint8_t* src_ptr = reinterpret_cast<const uint8_t*>(src.data()) + src_start * stride;
```

Comment:
tensor can hold GPU memory.
maybe we need to add a check that low bit type are supported only for non-RemoteTensors?

Reply:
Thanks for your advice,  remoteTensor check is added